biopet-core/src/main/scala/nl/lumc/sasc/biopet/core/BiopetQScript.scala

```diff
@@ -177,4 +177,52 @@ object BiopetQScript {
     require(outputDir.getAbsoluteFile.canWrite, s"No premision to write outputdir: $outputDir")
   }
 
+  def safeInputs(function: QFunction): Seq[File] = {
+    try {
+      function.inputs
+    } catch {
+      case e: NullPointerException => Seq()
+    }
+  }
+
+  def safeOutputs(function: QFunction): Seq[File] = {
+    try {
+      function.outputs
+    } catch {
+      case e: NullPointerException => Seq()
+    }
+  }
+
+  def safeDoneFiles(function: QFunction): Seq[File] = {
+    try {
+      function.doneOutputs
+    } catch {
+      case e: NullPointerException => Seq()
+    }
+  }
+
+  def safeFailFiles(function: QFunction): Seq[File] = {
+    try {
+      function.failOutputs
+    } catch {
+      case e: NullPointerException => Seq()
+    }
+  }
+
+  def safeIsDone(function: QFunction): Boolean = {
```

Comment:
I would change this function and `safeIsFail` to have type `Option[Boolean]`. You return simply `false` here when the underlying method does not exist, but I believe that's potentially giving a wrong answer. Returning `None` (as in, no boolean exists here) is semantically better, imho.